Tests/MockingKitTests/MockableAsyncTests.swift

```diff
@@ -0,0 +1,295 @@
+//
+//  MockableAsyncTests.swift
+//
+//  Created by Tobias Boogh on 2022-05-04.
+//
+
+import XCTest
+import Nimble
+@testable import MockingKit
+
+class MockableAsyncTests: XCTestCase {
+
+    fileprivate var mock: TestClass!
+
+    override func setUp() {
+        mock = TestClass()
+    }
+
+    func testRegisteringResult_registersFunctionWithReferenceId() {
+        let ref = mock.functionWithIntResultRef
+        let result: (String, Int) throws -> Int = { _, int in int * 2 }
+        mock.registerResult(for: ref, result: result)
```

Comment:
Would it be easier to use these new async utils and not require type definitions when e.g. defining the result function, if the new functions were called e.g. `registerAsyncResult`?

I haven't tried this yet, but perhaps that would make the compiler understand that any inline result block is throwing without having to specify it when defining the block?